src/services/completions.ts

```diff
@@ -585,9 +592,11 @@ namespace ts.Completions {
             }
 
             const { name, needsConvertPropertyAccess } = info;
+            const sortText = symbolToSortTextMap && symbolToSortTextMap[getSymbolId(symbol)]

```

Comment:
I think instead of spreading this code everywhere, (isDeprecatedSymbol), we should keep it same and change to deprecated sortText here. 
One change would be that we do is have SortTextId as number enum instead of `string`
```ts
const sortTextId = symbolToSortTextMap && symbolToSortTextMap[getSymbolId(symbol)] || SortTextId.LocationPriority;
const sortText = isDeprecated(symbol) ?  DeprecatedStart + sortTextId : sortTextId;
```

Reply:
@sheetalkamat Thanks for the feedback. Not sure that I fully understood the idea. Correct me if I'm wrong., so need to do the following changes

1. `sortText` property in `CompletionEntry` need to change from `string` to `number` - is it ok to TSServer response?
2. Create a helper function to avoid `isDeprecated` conditions, and use it for all sort types., not only in `getCompletionEntriesFromSymbols`

Reply:
No the type of property in completion entry does not change but sortTextMap use number enum and then convert those to string in one place without having to worry about deprecated sortText everywhere else.

Reply:
@sheetalkamat It looks like in `completions` there are places where `sortText` needs to be explicitly converted, like `createCompletionEntryForLiteral`, `getJSCompletionEntries`, etc. Is it ok to convert numbers to string in all of these cases?

Reply:
I am not suggesting changing type of `CompletionEntry["sortText"]` type. I am suggesting that `SymbolSortTextMap` is enum with new enum that has entries without deperecated and numbers so you can do toString() and convert to actual sortText enum here depending on deprecated.

Reply:
@@sheetalkamat I didn't mean changing type of `CompletionEntry["sortText"]`. I mean, if we change the `SortText` to numbers then we need to explicitly convert the `sortText` in cases like this `sortText: SortText.JavascriptIdentifiers.toString();`.

I've pushed the draft to be sure that we on the same page because I'm not sure about the correctness of using the new enum for SortText.